cadctl/cmd/cluster-missing/cluster-missing.go

```diff
@@ -29,45 +30,67 @@ import (
-		AWS_SECRET_ACCESS_KEY, hasAWS_SECRET_ACCESS_KEY := os.LookupEnv("AWS_SECRET_ACCESS_KEY")
-		AWS_SESSION_TOKEN, hasAWS_SESSION_TOKEN := os.LookupEnv("AWS_SESSION_TOKEN")
-		AWS_DEFAULT_REGION, hasAWS_DEFAULT_REGION := os.LookupEnv("AWS_DEFAULT_REGION")
-		if !hasAWS_ACCESS_KEY_ID || !hasAWS_SECRET_ACCESS_KEY || !hasAWS_SESSION_TOKEN || !hasAWS_DEFAULT_REGION {
-			return fmt.Errorf("one of the required envvars in the list '(AWS_ACCESS_KEY_ID AWS_SECRET_ACCESS_KEY AWS_SESSION_TOKEN AWS_DEFAULT_REGION)' is missing")
-		}
-
-		_, err := aws.NewClient(AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY, AWS_SESSION_TOKEN, AWS_DEFAULT_REGION)
-		if err != nil {
-			return fmt.Errorf("could not start awsClient: %w", err)
-		}
-
-		// in this case it's ok if the envvar is empty
-		CAD_OCM_FILE_PATH := os.Getenv("CAD_OCM_FILE_PATH")
-		_, err = ocm.New(CAD_OCM_FILE_PATH)
-		if err != nil {
-			return fmt.Errorf("could not create ocm client: %w", err)
-		}
-
-		CAD_PD, ok := os.LookupEnv("CAD_PD")
-		if !ok {
-			return fmt.Errorf("could not load CAD_PD envvar")
-		}
-
-		_, err = pagerduty.NewWithToken(CAD_PD)
-		if err != nil {
-			return fmt.Errorf("could not start pagerdutyClient: %w", err)
-		}
-
-		return nil
-	},
+	RunE:  run,
+}
+
+func run(cmd *cobra.Command, args []string) error {
+
+	AWS_ACCESS_KEY_ID, hasAWS_ACCESS_KEY_ID := os.LookupEnv("AWS_ACCESS_KEY_ID")
+	AWS_SECRET_ACCESS_KEY, hasAWS_SECRET_ACCESS_KEY := os.LookupEnv("AWS_SECRET_ACCESS_KEY")
+	AWS_SESSION_TOKEN, hasAWS_SESSION_TOKEN := os.LookupEnv("AWS_SESSION_TOKEN")
+	AWS_DEFAULT_REGION, hasAWS_DEFAULT_REGION := os.LookupEnv("AWS_DEFAULT_REGION")
+	if !hasAWS_ACCESS_KEY_ID || !hasAWS_SECRET_ACCESS_KEY || !hasAWS_SESSION_TOKEN || !hasAWS_DEFAULT_REGION {
+		return fmt.Errorf("one of the required envvars in the list '(AWS_ACCESS_KEY_ID AWS_SECRET_ACCESS_KEY AWS_SESSION_TOKEN AWS_DEFAULT_REGION)' is missing")
+	}
+
+	awsClient, err := aws.NewClient(AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY, AWS_SESSION_TOKEN, AWS_DEFAULT_REGION)
+	if err != nil {
+		return fmt.Errorf("could not start awsClient: %w", err)
+	}
+
+	// in this case it's ok if the envvar is empty
+	CAD_OCM_FILE_PATH := os.Getenv("CAD_OCM_FILE_PATH")
+	ocmClient, err := ocm.New(CAD_OCM_FILE_PATH)
+	if err != nil {
+		return fmt.Errorf("could not create ocm client: %w", err)
+	}
+
+	CAD_PD, ok := os.LookupEnv("CAD_PD")
+	if !ok {
+		return fmt.Errorf("could not load CAD_PD envvar")
+	}
+
+	_, err = pagerduty.NewWithToken(CAD_PD)
+	if err != nil {
+		return fmt.Errorf("could not start pagerdutyClient: %w", err)
+	}
+
+	chgmClient := chgm.Client{
+		ChgmService: chgm.ChgmProvider{
+			AwsClient: awsClient,
+			OcmClient: ocmClient,
+		},
+	}
+
+	res, err := chgmClient.AreAllInstancesRunning(externalClusterID)
+	if err != nil {
+		return fmt.Errorf("could not verify all instances were running: %w", err)
+	}
+	fmt.Printf("yay! the result is %#v", res)
+
+	return nil
 }
 
 var (
-	incidentID string
+	externalClusterID string = ""
```

Comment:
You can change this t o `externalClusterID string`. The initialization value of a string is the empty string. No need to set this.

Reply:
It's just so it's clear this is the default value :)

Reply:
Added a comment, if you like it a can resolve

Reply:
typo: `s/deafault/default/`

Reply:
K :)